controllers/configmap.go

```diff
@@ -33,18 +40,66 @@ func (r *RolloutManagerReconciler) reconcileConfigMap(ctx context.Context, cr ro
 
 	setRolloutsLabelsAndAnnotationsToObject(&desiredConfigMap.ObjectMeta, cr)
 
-	trafficRouterPlugins := []pluginItem{
-		{
+	trafficRouterPluginsMap := map[string]pluginItem{
+		OpenShiftRolloutPluginName: {
 			Name:     OpenShiftRolloutPluginName,
 			Location: r.OpenShiftRoutePluginLocation,
 		},
 	}
-	pluginString, err := yaml.Marshal(trafficRouterPlugins)
+
+	// Append traffic management plugins specified in RolloutManager CR
+	for _, plugin := range cr.Spec.Plugins.TrafficManagement {
+		// Prevent adding or modifying the OpenShiftRoutePluginName through the CR
+		if plugin.Name == OpenShiftRolloutPluginName {
+			return fmt.Errorf("the plugin %s cannot be modified or added through the RolloutManager CR", OpenShiftRolloutPluginName)
+		}
+		// Check for duplicate traffic plugins
+		if _, exists := trafficRouterPluginsMap[plugin.Name]; !exists {
+			trafficRouterPluginsMap[plugin.Name] = pluginItem{
+				Name:     plugin.Name,
+				Location: plugin.Location,
+			}
+		}
+	}
+
+	// Convert traffic plugins map to slice
+	trafficRouterPlugins := make([]pluginItem, 0, len(trafficRouterPluginsMap))
```

Comment:
can trafficRouterPluginsMap be initialized as slice itself, instead of map and then converting it into slice?

Reply:
Yes we can directly add data into slice but I am checking for duplicates here using map https://github.com/argoproj-labs/argo-rollouts-manager/blob/74e767e03f7096dcd58a05222e1a6909b067394e/controllers/configmap.go#L57 ,  so that user cannot add duplicate data through CR